app/javascript/components/patient/history/History.js

```diff
@@ -0,0 +1,193 @@
+
+  handleEditSubmit = () => {
+    this.edit();
+  };
+
+  delete = async () => {
+    const confirmText = 'Are you sure you would like to delete this comment? This action cannot be undone.';
+    const options = {
+      title: 'History',
+      okLabel: 'Delete',
+      okVariant: 'danger',
+      cancelLabel: 'Cancel',
+    };
+    if (await confirmDialog(confirmText, options)) {
+      this.handleDeleteSubmit();
+    }
+  };
+
+  handleDeleteClick = () => {
+    this.delete();
+  };
+
+  handleDeleteSubmit = () => {
+    this.setState({ loading: true }, () => {
+      axios.defaults.headers.common['X-CSRF-Token'] = this.props.authenticity_token;
+      axios
+        .delete(window.BASE_PATH + '/histories/' + this.props.history.id)
+        .then(() => {
+          location.reload(true);
+        })
+        .catch(error => {
+          reportError(error);
+        });
+    });
+  };
+
+  renderEditMode() {
+    return (
+      <React.Fragment>
+        <textarea
+          id="comment"
+          name="comment"
+          className="form-control"
+          aria-label="Edit comment input"
+          style={{ resize: 'none' }}
+          rows="3"
+          value={this.state.comment}
+          onChange={this.handleChange}
+        />
+        <Button
+          id="update-edit-history-btn"
+          variant="primary"
+          size="sm"
+          className="float-right mt-2"
+          disabled={this.state.loading || this.state.comment === ''}
+          onClick={this.handleEditSubmit}
+          aria-label="Submit Edit History Comment">
+          Update
+        </Button>
+        <Button
+          id="cancel-edit-history-btn"
+          variant="primary"
+          size="sm"
+          className="float-right mt-2 mr-2"
+          disabled={this.state.loading}
+          onClick={this.handleEditCancel}
+          aria-label="Cancel Edit History Comment">
+          Cancel
+        </Button>
+      </React.Fragment>
+    );
+  }
+
+  renderActionButtons() {
+    return (
+      <Col>
+        <div className="float-right" style={{ width: '45px' }}>
+          <span data-for={`edit-history-item-${this.props.history.id}`} data-tip="">
+            <Button id="edit-history-btn" variant="link" className="icon-btn p-0 mr-1" onClick={this.handleEditClick} aria-label="Edit History Comment">
+              <i className="fas fa-edit"></i>
+            </Button>
+          </span>
+          <ReactTooltip
+            id={`edit-history-item-${this.props.history.id}`}
+            multiline={true}
+            place="left"
+            type="dark"
+            effect="solid"
+            className="tooltip-container">
+            <span>You may edit comments you have added</span>
+          </ReactTooltip>
+          <span data-for={`delete-history-item-${this.props.history.id}`} data-tip="">
+            <Button id="delete-history-btn" variant="link" className="icon-btn p-0" onClick={this.handleDeleteClick} aria-label="Delete History Comment">
```

Comment:
if `handleDeleteClick` is just calling `this.delete` then why not change the `onClick` action to `() => this.delete()`.

Reply:
This comment is obsolete